corehq/apps/locations/models.py

```diff
@@ -436,9 +444,15 @@ def __init__(self, *args, **kwargs):
         if location_type:
             self.location_type = location_type
 
+    def __unicode__(self):
+        return u"{} ({})".format(self.name, self.domain)
+
     def __repr__(self):
-        val = u"%s (%s)" % (self.name, self.location_type)
-        return val.encode('utf-8')
+        return u"Location(domain='{}', name='{}', location_type='{}')".format(
```

Comment:
Totally.  Luckily, `__repr__` is one thing for which there *is* a [strong convention](https://docs.python.org/2/reference/datamodel.html#object.__repr__), which we should follow more than we do.
>  If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment). If this is not possible, a string of the form <...some useful description...> should be returned.